receiver/awscontainerinsightreceiver/internal/cadvisor/container_info_processor.go

```diff
@@ -136,16 +138,23 @@ func processContainer(info *cInfo.ContainerInfo, mInfo extractors.CPUMemInfoProv
 		tags[ci.PodIDKey] = podID
 		tags[ci.K8sPodNameKey] = podName
 		tags[ci.K8sNamespace] = namespace
-		if containerName != infraContainerName {
+		switch containerName {
+		// For docker, pause container name is set to POD while containerd does not set it.
+		// See https://github.com/aws/amazon-cloudwatch-agent/issues/188
+		case "", infraContainerName:
+			// NOTE: the pod here is only used by NetMetricExtractor,
+			// other pod info like CPU, Mem are dealt within in processPod.
+			containerType = ci.TypeInfraContainer
+		default:
 			tags[ci.ContainerNamekey] = containerName
 			containerID := path.Base(info.Name)
 			tags[ci.ContainerIDkey] = containerID
 			pKey.containerIds = []string{containerID}
 			containerType = ci.TypeContainer
-		} else {
-			// NOTE: the pod here is only used by NetMetricExtractor,
-			// other pod info like CPU, Mem are dealt within in processPod.
-			containerType = ci.TypePod
+			// TODO(pvasir): wait for upstream fix https://github.com/google/cadvisor/issues/2785
+			if !info.Spec.HasFilesystem {
+				logger.Debug("D! containerd does not have container filesystem metrics from cadvisor, See https://github.com/google/cadvisor/issues/2785")
```

Comment:
If this case happen, it will print the same message every time when the function is called. This might overwhelm the logs. Can we choose to print only a sample of it (like 1 out of 100 or 1000)? See https://github.com/uber-go/zap/blob/master/FAQ.md#why-sample-application-logs